inductiva/utils/format_utils.py

```diff
@@ -40,55 +39,49 @@ def seconds_formatter(secs: float) -> str:
     return str(datetime.timedelta(seconds=round(secs)))
 
 
+#clean this function
+def apply_formatters(rows, columns, formatters):
+    data = {}
+
+    for index, column_name in enumerate(columns):
+        data[column_name] = [row[index] for row in rows]
+
+    for column_name, formatter in formatters.items():
+        if column_name in data:
+            data[column_name] = [formatter(x) for x in data[column_name]]
+
+    return data
+
+
 def get_tabular_str(
     rows,
     columns,
-    default_col_space=15,
-    override_col_space=None,
     formatters=None,
 ) -> str:
     """Converts a list of lists to a string table.
 
-    Temporary solution to display tables. `pandas` dependency to print tables
-    is overkill, but since we have `pandas` anyway for now, we can use it.
     """
-    df = pd.DataFrame(rows, columns=columns)
-    formatters = formatters or {}
 
-    col_space = {col: default_col_space for col in columns}
-    col_space.update(override_col_space or {})
+    formatters = formatters or {}
 
-    # replace None with np.nan so that pandas can format them as "n/a"
-    # by passing na_rep="n/a" to to_string()
-    df.fillna(np.nan, inplace=True)
+    data = apply_formatters(rows, columns, formatters)
+    data_tabulated = tabulate(data, headers=columns, missingval="n/a")
 
-    return df.to_string(
-        index=False,
-        na_rep="n/a",
-        formatters=formatters,
-        col_space=col_space,
-    )
+    return data_tabulated
```

Comment:
the method name says something about getting a string but we're returning a tabulate object. Please return a string.

```suggestion
    return str(data_tabulated)
```

Reply:
Tabulate returns a string. I've changed the var data_tabulated to data_tabulated_str to be more clear.